velox/functions/sparksql/String.h

```diff
@@ -293,6 +292,62 @@ struct EndsWithFunction {
   }
 };
 
+/// locate function
+/// locate(string, string) -> integer
+/// locate(string, string, integer) -> integer
+///
+/// Returns the position of the first occurrence of the first string in the
+/// second string after the give position.
+template <typename T>
+struct LocateFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE bool call(
+      out_type<int32_t>& result,
+      const arg_type<Varchar>& subString,
+      const arg_type<Varchar>& string) {
+    result = stringImpl::stringPosition<false /*isAscii*/>(
+        string, subString, 1 /*instance*/, 1 /*start*/);
+    return true;
+  }
+
+  FOLLY_ALWAYS_INLINE bool callAscii(
+      out_type<int32_t>& result,
+      const arg_type<Varchar>& subString,
+      const arg_type<Varchar>& string) {
+    result = stringImpl::stringPosition<true /*isAscii*/>(
+        string, subString, 1 /*instance*/, 1 /*start*/);
+    return true;
+  }
+
+  FOLLY_ALWAYS_INLINE bool callAscii(
+      out_type<int32_t>& result,
+      const arg_type<Varchar>& subString,
+      const arg_type<Varchar>& string,
+      const arg_type<int32_t>& start) {
+    result = stringImpl::stringPosition<true /*isAscii*/>(
```

Comment:
maybe it's better to call `findNthInstanceByteIndexFromStart` here, which support start position parameter.

Reply:
`stringPosition` calls `findNthInstanceByteIndexFromStart`, and it also handles the edge cases like empty substr, invalid start, so we can avoid duplicate code.

Reply:
I see your point